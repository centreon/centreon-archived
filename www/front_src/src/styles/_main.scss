@import "partials/typography";
@import "partials/fonts";
@import "partials/variables";
@import "partials/messages";
@import "partials/icons";
@import "partials/form/form";
@import "partials/helpers";
@import "partials/mixins";
