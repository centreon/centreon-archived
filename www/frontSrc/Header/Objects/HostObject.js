import React from 'react'
import Button from 'material-ui/Button'
import Grid from 'material-ui/Grid'
import { withStyles } from 'material-ui/styles'
import numeral from 'numeral'
import Avatar from 'material-ui/Avatar'
import Popover from 'material-ui/Popover'
import Typography from 'material-ui/Typography'
import HostIcon from "../icons/HostIcon"

const styles = theme => ({
  root: {
    position: 'relative',
    margin: '0 12px',
    '&:before': {
      width: 1,
      height: 30,
      backgroundColor: '#d1d2d4',
      content: '""',
      position: 'absolute',
      top: 18,
      margin: '0px -10px',
    },
  },
  'a': {
    color: '#0072CE',
    '&:hover': {
      backgroundColor: '#064166'
    },
    '&:visited': {
      color: '#10069F'
    }
  },
  status: {
    margin: '4px',
    color: '#fff',
    width: 38,
    height: 38,
    '& span': {
      fontSize: 16
    },
  },
  errorStatus: {
    margin: '10px 4px',
    width: 46,
    height: 46,
    backgroundColor: theme.palette.error.main,
    '& span': {
      fontSize: 16,
      color: '#fff',
      fontWeight: '600'
    },
    '&:hover': {
      backgroundColor: theme.palette.error.light,
    },
  },
  unreachableStatus: {
    margin: '10px 4px',
    color: '#fff',
    width: 40,
    height: 40,
    '& span': {
      fontSize: 15,
      fontWeight: '600'
    },
    backgroundColor: theme.palette.unreachable.main,
    '&:hover': {
      backgroundColor: theme.palette.unreachable.light,
    },
  },
  chip: {
    height: '8px',
    width: '8px',
    borderRadius: 20,
    display: 'inline-table',
    marginRight: 6,
  },
  pendingStatus: {
    height: '8px',
    width: '8px',
    position: 'absolute',
    top: 42,
    left: 7,
    borderRadius: 20,
    backgroundColor: theme.palette.pending.main
  },
  icon: {
    width: 34,
      height: 34,
      display: 'inline-flex',
      verticalAlign: 'middle',
      margin: '6px',
      color: '#A7A9AC',
      cursor: 'pointer',
      '&:hover': {
      color: '#D1D2D4',
    }
  },
  paper: {
    padding: theme.spacing.unit,
  },
  popover: {
    pointerEvents: 'none',
  },
  objectDetails: {
    padding: '10px 16px',
    borderBottom: '1px solid #d1d2d4',
    '&:last-child' : {
      borderBottom: 'none',
    }
  },
  total: {
    float: 'right',
    marginLeft: 34,
  },
})

const HostObject = (
  {classes, object, anchorEl, open, handleClose, handleOpen,
    down, warning, unreachable, ok, pending, total, url
  }) => (
  <div className={classes.root}>
    <HostIcon
      viewBox="6 156 600 600"
      className={classes.icon}
      aria-haspopup="true"
      onClick={handleOpen}
    />
    {pending.total > 0 ?
      <span className={classes.pendingStatus} ></span>
      : ''
    }

    <Button variant="fab" href={down.url}
            className={(classes.status, classes.errorStatus)}>
      {numeral(down.unhandled).format('0a')}
    </Button>
    <Button variant="fab" mini href={unreachable.url}
            className={( classes.status, classes.unreachableStatus)}>
      {numeral(unreachable.unhandled).format('0a')}
    </Button>
    <Popover
      open={open}
      anchorEl={anchorEl}
      anchorReference='anchorEl'
      anchorPosition={{ top: 500, left: 400 }}
      onClose={handleClose}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'left',
      }}
      transformOrigin={{
        vertical: 'top',
        horizontal: 'left',
      }}
    >
      <div className={classes.objectDetails}>
        <Typography variant="caption" gutterBottom>
          <a href={url} title="all hosts list">
            All hosts
          </a>
          <span className={classes.total}>{total}</span>
        </Typography>
      </div>
      <div className={classes.objectDetails}>
        <Typography variant="caption" gutterBottom>
          <span className={classes.chip} style={{backgroundColor: '#e00b3d'}}></span>
          <a href={down.url} title="hosts down list">
            {down.unhandled} Unhandled problems
          </a>
          <span className={classes.total}> / {down.total}</span>
        </Typography>
      </div>
      <div className={classes.objectDetails}>
        <Typography variant="caption" gutterBottom>
          <span className={classes.chip} style={{backgroundColor: '#818285'}}></span>
          <a href={unreachable.url} title="hosts unreachable list">
            {unreachable.unhandled} Unreachable hosts
          </a>
          <span className={classes.total}> / {unreachable.total}</span>
        </Typography>
      </div>
      <div className={classes.objectDetails}>
        <Typography variant="caption" gutterBottom>
          <span className={classes.chip} style={{backgroundColor: '#88b917'}}></span>
          <a href={ok.url} title="hosts ok list">
            {ok.total} Ok hosts
          </a>
        </Typography>
      </div>
      {pending.total > 0 ?
        <div className={classes.objectDetails}>
          <Typography variant="caption" gutterBottom>
            <span className={classes.chip} style={{backgroundColor: '#2AD1D4'}}></span>
            <a htef={pending.url} title="pending hosts list">
            {pending.total} Pending hosts
            </a>
          </Typography>
        </div> : ''
      }
    </Popover>
  </div>
)

export default withStyles(styles)(HostObject)
