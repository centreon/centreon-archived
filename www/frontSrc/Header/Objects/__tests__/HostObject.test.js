/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import HostObject from '../HostObjectContainer'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'

Enzyme.configure({ adapter: new Adapter() })

const theme = createMuiTheme({
  palette: {
    primary: {main: '#88b917', dark: '#597F00'},
    secondary: {main: '#00bfb3'},
    error: {main: '#e00b3d', light: '#ff2c5b', lighter: '#FF8C88'},
    warning: {main: '#ff9a13', light: '#ffb749'},
    unreachable: {main: '#818285', light: '#a6a7a6'},
    unknown: {main: '#bcbdc0', dark: '#95969a'},
    pending: {main: '#2AD1D4'},
  },
  font: {
    openSans: "'Open Sans', Arial, Tahoma, Helvetica, Sans-Serif"
  },
})

describe('HostObject', () => {
  const middlewares = [ thunk ]
  const mockStore = configureStore(middlewares)

  const store = mockStore({
    host: {
      url: './main.php?p=20202&o=h&search=',
      down: {
        classe: 'error',
        total: '5',
        unhandled: '0',
        url: './main.php?p=20202&o=h_down&search='
       },
      ok: {
        classe: 'success',
        total: '58',
        url: './main.php?p=20202&o=h_up&search='
      },
      unreachable: {
        classe: 'unreachable',
        total: '0',
        unhandled: '0',
        url: './main.php?p=20202&o=h_unreachable&search='
      },
      pending: {
        classe: 'pending',
        total: '2',
        url: './main.php?p=20202&o=h_pending&search='
      },
      error: null,
      isFetching: true,
      refreshTime: 90000,
      total: 150,
      dataFetched: true
    }
  })

  let getHosts = jest.fn()

  it('host status renders correctly', () => {
    const component = shallow(<HostObject store={store} />)
    expect(toJson(component)).toMatchSnapshot()
  })


  it('should display detail popover when clicking on host icon', () => {
    const component = mount(
      <MuiThemeProvider theme={theme}>
        <HostObject store={store} getHosts={getHosts} />
      </MuiThemeProvider>
    )

    expect(component.find('div#hostPopover').length).toEqual(0)
    component.find('svg#hostIcon').simulate('click')
    expect(component.find('div#hostPopover').length).toEqual(1)
  })

})