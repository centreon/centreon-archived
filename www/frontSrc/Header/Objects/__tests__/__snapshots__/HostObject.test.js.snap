// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostObject host status renders correctly 1`] = `
<HostObjectContainer
  getHosts={[Function]}
  host={
    Object {
      "dataFetched": true,
      "down": Object {
        "classe": "error",
        "total": "5",
        "unhandled": "0",
        "url": "./main.php?p=20202&o=h_down&search=",
      },
      "error": null,
      "isFetching": true,
      "ok": Object {
        "classe": "success",
        "total": "58",
        "url": "./main.php?p=20202&o=h_up&search=",
      },
      "pending": Object {
        "classe": "pending",
        "total": "2",
        "url": "./main.php?p=20202&o=h_pending&search=",
      },
      "refreshTime": 90000,
      "total": 150,
      "unreachable": Object {
        "classe": "unreachable",
        "total": "0",
        "unhandled": "0",
        "url": "./main.php?p=20202&o=h_unreachable&search=",
      },
      "url": "./main.php?p=20202&o=h&search=",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
